core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -167,20 +170,29 @@ private class AsyncEventQueue(
     }
     logTrace(s"Dropping event $event")
 
-    val droppedCount = droppedEventsCounter.get
+    val droppedCount = droppedEventsCounter.get - lastDroppedEventsCounter
+    val lastReportTime = lastReportTimestamp.get
+    val curTime = System.currentTimeMillis()
     if (droppedCount > 0) {
       // Don't log too frequently
-      if (System.currentTimeMillis() - lastReportTimestamp >= 60 * 1000) {
-        // There may be multiple threads trying to decrease droppedEventsCounter.
-        // Use "compareAndSet" to make sure only one thread can win.
-        // And if another thread is increasing droppedEventsCounter, "compareAndSet" will fail and
-        // then that thread will update it.
-        if (droppedEventsCounter.compareAndSet(droppedCount, 0)) {
```

Comment:
So if I understand correctly, droppedEventsCounter can be changed between L170 to L178 even without contention of L178, as L161 allows other threads to increase the value as well. Do I understand correctly?

Reply:
@HeartSaVioR  Yes, that's what I mean. In some cases, the log can never be logged and it's hard to know what happened that may causing it slow.

Reply:
Can we just remove this check and use `droppedEventsCounter.getAndSet()` instead?